lib/posix-process/arch/x86_64/execve.c

```diff
@@ -0,0 +1,38 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2024, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+#include <uk/arch/ctx.h>
+#include <uk/essentials.h>
+
+void execve_arch_execenv_init(struct ukarch_execenv *execenv_new,
+			      struct ukarch_execenv *execenv,
+			      __uptr ip, __uptr sp)
+{
+	UK_ASSERT(execenv_new);
+	UK_ASSERT(execenv);
+	UK_ASSERT(ip);
+	UK_ASSERT(sp);
+	UK_ASSERT(IS_ALIGNED(sp, UKARCH_SP_ALIGN));
+
+	execenv_new->regs.rip = ip;
```

Comment:
Where do you init the other gp regs? Ditto for ARM.

Reply:
I think it's okay to leave gprets into an undefined state, as we jump into a new execution context.